### DISCUSSION TITLE
Identify common way of how to handle baselines

### DISCUSSION THREAD

**Discussion by zeuchste** [nebulastream/nebulastream#11] [Ideas]
Currently, the code is full of baseline implementations. One example is we have two query managers, a dynamic one that is the default and a static one that we mainly implemented to have a baseline for the paper. What is our general take here? Possible solutions are

1. We remove all baseline implementations
2. We hide baseline implementation behind plugins/marcros or so and do not maintain them
3. We leave both in and maintain both

**Comment by keyseven123** [comment]
For the first milestone, I really want to remove anything that is not necessary to keep a repo as lean as possible. Therefore, I would vote for 1.

**Reply by zeuchste** [reply]
but to clarify, do you remove the Hash join then and only stick to the NL Join? where do we draw the border @keyseven123

**Reply by keyseven123** [reply]
I liked what @alepping wrote and will add everything that is not 100% needed or tested.

> I would remove everything that we are not actively using first. We should document what we removed, so we can readd it later. If we don't have a join with a bloom filter, then we don't need a bloom filter. In my opinion it is as simple as that.

_Originally posted by @alepping in https://github.com/nebulastream/nebulastream-public/discussions/4#discussioncomment-9703992_

To answer your question, I would keep the HJ and NLJ but remove bucketing and the different hash table variants.

**Reply by zeuchste** [reply]
mh this sounds for me very subjective and not really reproduceable but I am fine, but again can we file a list of this then somewhere such that we have all the things to be removed on the table?
